/**:
  ros__parameters:
    delay_compensation_time: 0.17

    enable_smooth_stop: true
    enable_overshoot_emergency: true
    enable_large_tracking_error_emergency: true
    enable_slope_compensation: true
    enable_keep_stopped_until_steer_convergence: true

    # state transition
    drive_state_stop_dist: 0.5
    drive_state_offset_stop_dist: 1.0
    stopping_state_stop_dist: 0.5
    stopped_state_entry_duration_time: 0.1
    stopped_state_entry_vel: 0.01
    stopped_state_entry_acc: 0.1
    emergency_state_overshoot_stop_dist: 1.5

    # drive state
    kp: 1.0
    ki: 0.1
    kd: 0.0
    max_out: 1.0
    min_out: -1.0
    max_p_effort: 1.0
    min_p_effort: -1.0
    max_i_effort: 0.3
    min_i_effort: -0.3
    max_d_effort: 0.0
    min_d_effort: 0.0
    lpf_vel_error_gain: 0.9
    enable_integration_at_low_speed: false
    current_vel_threshold_pid_integration: 0.5
    time_threshold_before_pid_integration: 2.0
    enable_brake_keeping_before_stop: false
    brake_keeping_acc: -0.2

    # smooth stop state
    smooth_stop_max_strong_acc: -0.5
    smooth_stop_min_strong_acc: -0.8
    smooth_stop_weak_acc: -0.3
    smooth_stop_weak_stop_acc: -0.8
    smooth_stop_strong_stop_acc: -3.4
    smooth_stop_max_fast_vel: 0.5
    smooth_stop_min_running_vel: 0.01
    smooth_stop_min_running_acc: 0.01
    smooth_stop_weak_stop_time: 0.8
    smooth_stop_weak_stop_dist: -0.3
    smooth_stop_strong_stop_dist: -0.5

    # stopped state
    stopped_vel: 0.0
    stopped_acc: -3.4 # denotes pedal position
    stopped_jerk: -5.0

    # emergency state
    emergency_vel: 0.0
    emergency_acc: -5.0 # denotes acceleration
    emergency_jerk: -3.0

    # acceleration feedback
    lpf_acc_error_gain: 0.98
    acc_feedback_gain: 0.0

    # acceleration limit
    max_acc: 3.0
    min_acc: -5.0

    # jerk limit
    max_jerk: 2.0
    min_jerk: -5.0
    max_acc_cmd_diff: 50.0 # [m/s^2 * s^-1]

    # slope compensation
    lpf_pitch_gain: 0.95
    slope_source: "trajectory_goal_adaptive" # raw_pitch, trajectory_pitch, trajectory_adaptive or trajectory_goal_adaptive
    adaptive_trajectory_velocity_th: 1.0
    max_pitch_rad: 0.1
    min_pitch_rad: -0.1
